#166789954 Users should be able to Login via Google Facebook or Twitter

elemanhillary-zz opened the pull request:
#### What does this PR do?
- adds social login buttons
#### Description of Task to be completed
- enables social login functionality
#### How should this be manually tested?
```
git clone https://github.com/andela/ah-92explorers-frontend.git
cd ah-92explorers-frontend
git ft-login-via-social-sites-166789954
configure .env file `BASE_URL` value
npm install
npm run dev
```
- Navigate to `http://localhost:8080/login`
#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
[#166789954](https://www.pivotaltracker.com/story/show/166789954)
#### Screenshots (if appropriate)
![Imgur](https://i.imgur.com/mtosrB9.png)
#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

Commit:
feat(user-login): users should be able to Login

- add a login route
- add login component
- add login container
- validate inputs
- consume login endpoint from the backend
- manage states
- redirect routes
- add tests

[Start #166789953]

Commit:
feat(user-login): users should be able to Login

- add a login route
- add login component
- add login container
- validate inputs
- consume login endpoint from the backend
- manage states
- redirect routes
- add tests

[Start #166789953]

Hervera reviewed (changes requested):
Fix the circleCI build error

IsaiahRn reviewed (changes requested):
From the screenshot that you have provided, the twitter icon is missing.
Kindly, add that icon?

IsaiahRn commented on src/__tests__/__actions__/socialLogin.spec.js:
```diff
@@ -0,0 +1,123 @@
+    // global.window = Object.create(window);
+    const url = '?token=0f362c370ad9d1d4546c1ac6c8e36e94dfa0ef81ce71e1dabc22753b86d30b6feb7b6432773ebbd5cdcc329a0ef523bce5050aea4d79bb809c56683aae2ab609971b994d920499e1b2b25afc98d2c4f1a2451ca7931695eec685aeede7c37b2302bdd4cebcbbe35a04eaab8cccc73ef015109976fda2b99263d64f35fa79e1adba449772e5901994a9913ba071dc229a06a96f6f2a118e148d7a02139f6019198c11319ab32043d417e767258d677a8a005473835be1c33a0cccd8f ...
+    // windowSpy.mockImplementation(() => ({
+    //   location: {
```
Can you remove all these commented lines of code?

elemanhillary committed:
ft(social-login): users should be able to login via google facebook or twitter
- enables login using social networks(google, facebook, twitter)

[Finishes #166789954]

 Users Should Be Able Signup To Authors Haven

Hervera reviewed (changes requested):
Remove conflicts

kodek-sleuth commented on src/__tests__/components/social.spec.js:
```diff
@@ -0,0 +1,19 @@
+    const wrapper = mount(<Provider store={store}><Social /></Provider>);
+    const btn = wrapper.find('.twitter').first();
+    btn.simulate('click');
+    expect(btn.length).toBe(1);
```
I have realised that you not actually testing what the button will do after stimulation. This is universal for All the three buttons, Please kindly indicate those tests there

kodek-sleuth commented on src/components/continue/continue.js:
```diff
@@ -0,0 +1,37 @@
+import React, { Component, Fragment } from 'react';
```
The folder name should start with an Uppercase letter and all files with `jsx` should have the `jsx` extension. This is based on The Team Coding Conventions we signed up for

elemanhillary-zz commented on src/components/social/social.js:
```diff
@@ -0,0 +1,45 @@
+class Social extends Component {
+  handleOnClick = (e) => {
+    const { social: redirect } = this.props;
+    const classNames = e.target.className.split(' ');
```
use a unique` id` for each of the buttons.

elemanhillary-zz commented on src/components/social/social.js:
```diff
@@ -0,0 +1,45 @@
+const { BASE_URL } = process.env;
+class Social extends Component {
+  handleOnClick = (e) => {
+    const { social: redirect } = this.props;
```
rename to `redirectURL`

elemanhillary-zz commented on src/redux/actions/actionCreators/social.action.js:
```diff
@@ -0,0 +1,52 @@
+    return dispatch(failure(error));
+  }
+  try {
+    const data = dec(tokenStr);
```
use `async/await` here

elemanhillary committed:
ft(social-login): incoporate changes from develop branch
- users should be able to login with social accounts

[Finishes #166789954]

IsaiahRn reviewed (approved):
It can be merged

Hervera reviewed (approved):
Ready to be merged
